### PR TITLE
13 comment count add to get art by

### PR DESCRIPTION
**Pull request by MGough22**
feature, add comment count to get article by article id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -379,6 +380,7 @@ describe("PATCH /api/articles/:article_id", () => {
         const {
           body: { article },
         } = await request(app).get(`/api/articles/${testArticleId}`);
+        delete article.comment_count;
```
Comment (nc-nathan-blanch) [inline __tests__/app.test.js:383]: what's this line doing?

File: models/api.model.js [hunk 2]
```diff
@@ -5,9 +5,18 @@ exports.fetchTopics = async () => {
   return rows;
 };
 
+const sharedFetchArticleQueryLogic = `SELECT articles.*,
```
Comment (nc-nathan-blanch) [inline models/api.model.js:8]: seems unnecessary..

**Review by nc-nathan-blanch — COMMENTED** [review]
You should not be refactoring a previous test when you implement something new. This should have been a completely new test.

File: __tests__/app.test.js (hunk 1 above)
Comment (MGough22) [reply __tests__/app.test.js:383]: It's due to the way I was testing the accuracy of the return value in the patch endpoint: by comparing the updated value returned in the response, to one subsequently called directly from the db. The reason I then deleted this property from the object prior to comparing it, is that the feature (in this pr) changed the endpoint get/api/articles to add a comment_count to the response when called, but this is not present in the article table itself, so would result in a false positive in the prior comparison logic, without the amendment made on line 383.

**Pull request merged**